### ISSUE TITLE
Windows CPU hotplug

### ISSUE THREAD

**Issue by rbradford** [cloud-hypervisor/cloud-hypervisor#1799]
Hotplugged CPUs appear in Device Manager but don't appear in Resource Monitor until after a reboot.

![image](https://user-images.githubusercontent.com/588224/94909409-05485300-049b-11eb-90a7-7158a1d222ad.png)

**Comment by sboeuf** [comment]
Isn't it caused by something missing in the guest? We need to online the CPUs in Linux.

**Comment by rbradford** [comment]
@liuw @weltling Can you look into this?

**Comment by weltling** [comment]
I could look into this, would need some steps to incude the behavior.

Thanks.

**Comment by sboeuf** [comment]
@weltling 
Start the VM with the parameter `--cpus boot=1,max=4,kvm_hyperv=on` so that it boots with 1 vCPU, but leave you some room to extend up to 4 during runtime. Also, make sure you append the parameter `--api-socket=/tmp/ch.sock` so that you can access the API server socket.

Once the VM is up and running, use another shell to resize the number of vCPUs: 
```
./target/debug/ch-remote --api-socket=/tmp/ch.sock resize --cpus 4
```

**Comment by weltling** [comment]
I went to check how this currently is implemented in QEMU. It turns out, that QEMU functionality regarding Windows is identical. I used the command below to start a Windows guest:

`qemu-system-x86_64 -machine q35,accel=kvm -cpu host -smp 1,maxcpus=2 -m 4G -drive if=none,id=root,file=./windows.qcow -device virtio-blk-pci,drive=root,disable-legacy=on -vga std`

This sets the initial CPU amount in the guest to 1, while an additional CPU can be injected.

In the QEMU window, I then switched to the monitor console by pressing ctrl-alt-2 and there I typed in the commands below:

```
info hotplugable-cpus
device_add host-x86_64-cpu,id=cpu1,core-id=0,socket-id=1,thread-id=0
```

The result is same as in CH – the added CPU is visible in the device manager but is not usable until reboot.

Linux guests require a extra action inside the guest to online a hot added CPU

`echo 1 > /sys/devices/system/cpu/cpu1/online`

Thus, the next step would be to research what is the equivalent on Windows.

Thanks
